delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+            err_msg = "Failed to get storage from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_disk(self, storage_id):
+        try:
+            disk_info = self.ssh_do_exec([constant.DISK_INFO_COMMAND])
+            disk_map_list = []
+            self.split_value_map_list(
+                disk_info, disk_map_list, 'Capacity', split=":")
+            disks_list = []
+            for disk_map in disk_map_list:
+                if 'Status' in disk_map:
+                    size = disk_map['Capacity'].split("GiB")[0] + "GB"
+                    status = constants.DiskStatus.NORMAL \
+                        if disk_map['Status'] == 'OK' \
+                        else constants.DiskStatus.ABNORMAL
+                    type = disk_map['Type']
+                    type_array = type.split(";")
+                    model = vendor = version = None
+                    if len(type_array) > 2:
+                        model = type_array[1].replace('Model', '')
+                        vendor = type_array[0].replace('Make', '')
+                        version = type_array[2].replace('Revision', '')
+                    pool_id = disk_map.get('Usedinspan', None)
+                    serial_number = disk_map['Luid'].split(']')[-1]
+                    if pool_id:
+                        pool_id = pool_id.split('(')[0]
+                    disk_model = {
+                        'name': disk_map['HDSdevname'],
+                        'storage_id': storage_id,
+                        'native_disk_id': disk_map['DeviceID'],
+                        'serial_number': serial_number,
+                        'manufacturer': vendor,
+                        'model': model,
+                        'firmware': version,
+                        'speed': None,
+                        'capacity': int(self.get_size(size)),
+                        'status': status,
+                        'physical_type': None,
+                        'logical_type': None,
+                        'native_disk_group_id': pool_id,
+                        'location': disk_map['Serialnumber'],
+                    }
+                    disks_list.append(disk_model)
+            return disks_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get disk from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get disk from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_pool_size(self):
+        size_info = self.ssh_do_exec([constant.POOL_SIZE_COMMAND])
+        size_array = size_info.split("\r\n")
+        size_map = {}
+        pool_name = None
+        count = 0
```

Comment:
What does this `count` for? Seems not used.

Reply:
has deleted